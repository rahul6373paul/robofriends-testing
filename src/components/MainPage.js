import React, { Component } from 'react';
import { CardList } from '../components/CardList';
import SearchBox from '../components/SeachBox';
import Scroll from '../components/Scroll';
import Header from '../components/Header';

import './MainPage.css';

class MainPage extends Component {

    componentDidMount(){
      this.props.onReqRobots();
    }

    filteredRobots = () => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase()) ;
        })        
    }

    render(){

        return (this.props.isPending?         
                    <h1>Loading ...</h1> :        
                        (<div>
                            <div className="tc">
                                <Header />
                                <SearchBox searchChange={this.props.onSearchChange} />
                                <Scroll>
                                    <CardList robots={this.filteredRobots(this.props.robots)} />
                                </Scroll>
                            </div>
                        </div>)
        );
    
        }
    
    
}

export default MainPage;