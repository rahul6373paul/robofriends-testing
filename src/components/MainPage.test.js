import { shallow, mount, render,configure } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

let wrapper;
beforeEach( ()=> {
    const mockProps = {
        onReqRobots: jest.fn(),
        robots: [],
        searchField: "",
        isPending: false
    }
    wrapper = shallow(<MainPage {...mockProps} />)
})

it('expect to render MainPage Component', () => {
    expect(wrapper).toMatchSnapshot();    
});

it('filter robots correctly', () => {
    const mockProps2 = {
        onReqRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchField: "a",
        isPending: false
    }
    const wrapper2 = shallow(<MainPage {...mockProps2} />)
    expect(wrapper2.instance().filteredRobots()).toEqual([]);
    expect(wrapper2.instance().filteredRobots()).toEqual([]);

})