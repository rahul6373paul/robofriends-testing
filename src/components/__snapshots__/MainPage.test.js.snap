// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render MainPage Component 1`] = `
<div>
  <div
    className="tc"
  >
    <Header />
    <SearchBox />
    <Scroll>
      <CardList
        robots={Array []}
      />
    </Scroll>
  </div>
</div>
`;
